lib/src/clojure_lsp/queries.clj

```diff
@@ -319,12 +319,28 @@
       keyword-usage))
 
 (defmethod find-definition :var-definitions
-  [db {:keys [imported-ns] :as var-definition}]
-  (if (some #(safe-equal? 'potemkin/import-vars %) (defined-bys var-definition))
-    (find-definition db (assoc var-definition
-                               :bucket :var-usages
-                               :to imported-ns))
-    var-definition))
+  [db {:keys [imported-ns defined-by name ns] :as var-definition}]
+  (let [xf-declared-definition (comp xf-analysis->var-definitions
+                                     (xf-same-fqn ns name)
+                                     (filter #(not= 'clojure.core/declare (:defined-by %)))
+                                     (xf-same-lang var-definition))
+        actual-definition (find-last-order-by-project-analysis
+                            xf-declared-definition
+                            (db-with-ns-analysis db ns))]
```

Comment:
with this change we are doing this find for every single usage of `find-definition` feature, which is the most used clojure-lsp feature, where we only really want to do when the definied-by is a declare, I'm wondering if we could wrap this with a `delay` and only use it [here](https://github.com/clojure-lsp/clojure-lsp/pull/2025/files#diff-953310fb5e54c1b32a0f931450ceb70ffc4e0130927391c85964242ebbc61c7cR339) after the definied-by check, WDYT?